benchmark/partial_tweets/rapidjson_sax.h

```diff
@@ -0,0 +1,165 @@
+                else if (!(values & found_text) && (length == 4) && (memcmp(key,"text",4) == 0)) { values |= (key_text); }
+                else if (!(values & found_reply) && (length == 21) && (memcmp(key,"in_reply_to_status_id",21) == 0)) { values |= (key_reply); }
+                // Check if entering user object
+                else if ((length == 4) && (memcmp(key,"user",4) == 0)) { userobject_id = userobject_screen_name = true; }
+                // Must also check if in a user object
+                else if (!(values & found_userid) && userobject_id && (length == 2) && (memcmp(key,"id",2) == 0)) { values |= (key_userid); }
+                // Must also check if in a user object
+                else if (!(values & found_screenname) && userobject_screen_name && (length == 11) && (memcmp(key,"screen_name",11) == 0)) { values |= (key_screenname); }
+                else if (!(values & found_rt) && (length == 13) && (memcmp(key,"retweet_count",13) == 0)) { values |= (key_rt); }
+                else if (!(values & found_fav) && (length == 14) && (memcmp(key,"favorite_count",14) == 0)) { values |= (key_fav); }
+            }
+            else if ((length == 9) && (memcmp(key,"retweeted",9) == 0)) { inretweet = false; }  // Check if end of retweet
+            return true;
+        }
+        bool Uint(unsigned i) {
+            if (values & key_userid && !(values & found_userid)) {    // user.id
+                user_id = i;
+                userobject_id = false;
+                values &= ~(key_userid);
+                values |= (found_userid);
+            }
+            else if (values & key_rt && !(values & found_rt)) {   // retweet_count
+                rt = i;
+                values &= ~(key_rt);
+                values |= (found_rt);
+            }
+            else if (values & key_fav && !(values & found_fav)) {   // favorite_count
+                fav = i;
+                values &= ~(key_fav);
+                values |= (found_fav);
+                // Assume that this is last key required, so add the partial_tweet to result
+                result.emplace_back(partial_tweets::tweet<std::string_view>{
+                date,id,text,reply_status,{user_id,screen_name},rt,fav});
+            }
+            return true;
+        }
+        bool Uint64(uint64_t i) {
+            if (values & key_id && !(values & found_id)) {    // id
+                id = i;
+                values &= ~(key_id);
+                values |= (found_id);
+            }
+            else if (values & key_reply && !(values & found_reply)) {   // in_reply_status_id
+                reply_status = i;
+                values &= ~(key_reply);
+                values |= (found_reply);
+            }
+            return true;
+        }
+        bool String(const char* str, SizeType length, bool copy) {
+            if (values & key_date && !(values & found_date)) {   //  created_at
+                date = str;
```

Comment:
Nitpicking: I think you want `data = {str,length}` or something of the sort. Your current code is going to call `strlen(str)` but it is wasteful and possibly unsafe given that you have the `length` parameter right there.